### PR TITLE
feat(vertex_httpx.py): update usage metadata handling

### LINKED ISSUE #4559: [Bug]: vertex_httpx.py breaks on missing `usageMetadata` keys
### What happened?

https://cloud.google.com/vertex-ai/generative-ai/docs/reference/nodejs/latest/vertexai/usagemetadata

These fields in `usageMetadata` are optional and cannot be assumed to be present.

### Relevant log output

```shell
litellm.APIConnectionError: 'promptTokenCount'
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/litellm/utils.py", line 9668, in __anext__
    async for chunk in self.completion_stream:
  File "/usr/local/lib/python3.11/site-packages/litellm/llms/vertex_httpx.py", line 1416, in __anext__
    return self.chunk_parser(chunk=json_chunk)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/litellm/llms/vertex_httpx.py", line 1346, in chunk_parser
    prompt_tokens=processed_chunk["usageMetadata"]["promptTokenCount"],
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
KeyError: 'promptTokenCount'
```

### Twitter / LinkedIn details

https://twitter.com/DaveManouchehri

### PR DESCRIPTION
**Pull request by Manouchehri**
## Title

update usage metadata handling

## Relevant issues

Fixes #4559.

## Type

🐛 Bug Fix

## Changes

This commit updates the usage metadata handling in the `VertexLLM` and `ModelResponseIterator` classes. Instead of directly accessing the `usageMetadata` dictionary, the code now uses the `get` method to safely retrieve the values. This change ensures that the code gracefully handles cases where the `promptTokenCount`, `candidatesTokenCount`, or `totalTokenCount` keys are missing from the dictionary.

## [REQUIRED] Testing - Attach a screenshot of any new tests passing locally

Before commit:

```
dave@mbp hey % ./main.py
litellm.APIConnectionError: 'promptTokenCount'
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/litellm/utils.py", line 9668, in __anext__
    async for chunk in self.completion_stream:
  File "/usr/local/lib/python3.11/site-packages/litellm/llms/vertex_httpx.py", line 1416, in __anext__
    return self.chunk_parser(chunk=json_chunk)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/litellm/llms/vertex_httpx.py", line 1346, in chunk_parser
    prompt_tokens=processed_chunk["usageMetadata"]["promptTokenCount"],
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
KeyError: 'promptTokenCount'
```

After commit:

```
dave@mbp hey % ./main.py
Choice(delta=ChoiceDelta(content=None, function_call=None, role=None, tool_calls=None), finish_reason=None, index=0, logprobs=None)
1 + 1 = 2
```

### CODE REVIEW

File: litellm/llms/vertex_httpx.py [hunk 1]
```diff
@@ -845,9 +849,9 @@ def _get_token_and_url(
             endpoint = "generateContent"
             if stream is True:
                 endpoint = "streamGenerateContent"
-                url = f"https://{vertex_location}-aiplatform.googleapis.com/v1/projects/{vertex_project}/locations/{vertex_location}/publishers/google/models/{model}:{endpoint}?alt=sse"
+                url = f"https://{vertex_location}-aiplatform.googleapis.com/v1beta1/projects/{vertex_project}/locations/{vertex_location}/publishers/google/models/{model}:{endpoint}?alt=sse"
```
Comment (ghost) [inline litellm/llms/vertex_httpx.py:852]: Hey @Manouchehri, why is this change necessary?

**Comment by ghost** [comment]
Closing in favor of - https://github.com/BerriAI/litellm/commit/015a398713502d5b84c210b87bf106db277cdf3d

File: litellm/llms/vertex_httpx.py (hunk 1 above)
Comment (Manouchehri) [reply litellm/llms/vertex_httpx.py:852]: @krrishdholakia It's in the commit message:

Use v1beta1 endpoint, as it is needed for the cached_content feature.

Comment (ghost) [reply litellm/llms/vertex_httpx.py:852]: got it - can we use a flag for this? 

Google recommends using Vertex ai for prod

Google says v1beta endpoints can change without notice

Comment (Manouchehri) [reply litellm/llms/vertex_httpx.py:852]: Hmm, if you could do a check if `cached_content` is used, and then change the endpoint based on that, that would work I guess.
